crates/sui-indexer/src/store/pg_indexer_store_v2.rs

```diff
@@ -865,6 +950,38 @@ impl IndexerStoreV2 for PgIndexerStoreV2 {
         Ok(())
     }
 
+    async fn persist_object_snapshot(&self) -> Result<(), IndexerError> {
+        let latest_cp = self
+            .get_latest_tx_checkpoint_sequence_number()?
+            .unwrap_or_default();
+        let latest_snapshot_cp = self
+            .get_latest_object_snapshot_checkpoint_sequence_number()?
+            .unwrap_or_default();
+
+        if latest_cp <= latest_snapshot_cp + self.object_snapshot_max_checkpoint_lag as u64 {
```

Comment:
Can you add some comments explaining the relationship between `latest_cp`, `latest_snapshot_cp`, `object_snapshot_max_checkpoint_lag`. It's slightly hard for someone who doesn't know this code well to understand why we are returning here, and the logic in general. Also `object_snapshot_min_checkpoint_lag` seems to be only used for determining the window? What if we store the window length directly?

Reply:
> What if we store the window length directly?
yes, we can do constants of either max_lag & min_lag, or max_lag & window_size, I think min_lag is more meaningful for the clients, b/c this is the min size of the consistent read range, while window size is more of impl details